### PR TITLE
sensors: ccs811: Implement sampling mode switching

### PR DESCRIPTION
**Pull request by tkln**
CCS811 measurement mode selection is implemented with sensor attributes.
A new measurement interval sensor attribute is introduced.

### CODE REVIEW

File: drivers/sensor/ccs811/ccs811.c [hunk 1]
```diff
@@ -102,9 +102,46 @@ static int ccs811_channel_get(struct device *dev,
...
+	 */
+	if (val->val1 == 0 && val->val2	== 0) {
+		mode = CCS811_MODE_IDLE;
+	} else if (val->val1 == 1 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_1SEC;
+	} else if (val->val1 == 10 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_10SEC;
+	} else if (val->val1 == 60 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_60SEC;
+	} else {
```
Comment (Mani-Sadhasivam) [inline drivers/sensor/ccs811/ccs811.c]: Not implementing `CCS811_MODE_RAW_DATA`? It might be useful in some cases.

**Review by Mani-Sadhasivam — COMMENTED** [review]
Hi @tkln,

Overall it looks good. Just one query inline.

**Comment by tkln** [comment]
@Mani-Sadhasivam I've not implemented switching to Mode 4 because SENSOR_CHAN_CO2 and SENSOR_CHAN_VOC readings cannot be used in Mode 4. I'm mostly just interested in changing the measurement mode due to the lower power consumption of the other modes.

**Comment by Mani-Sadhasivam** [comment]
@tkln SInce we support `CURRENT` and `VOLTAGE` measurements in the driver, the corresponding mode should also be enabled. I can't figure out a valid usecase for this but it would be good to have.

**Comment by tkln** [comment]
@Mani-Sadhasivam I've now added Mode 4 and also made the polling delays to adjust accordingly with the measurement time.

File: drivers/sensor/ccs811/ccs811.c [hunk 2]
```diff
@@ -102,9 +115,55 @@ static int ccs811_channel_get(struct device *dev,
...
+	if (val->val1 == 0 && val->val2	== 0) {
+		mode = CCS811_MODE_IDLE;
+	} else if (val->val1 == 1 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_1SEC;
+	} else if (val->val1 == 10 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_10SEC;
+	} else if (val->val1 == 60 && val->val2 == 0) {
+		mode = CCS811_MODE_IAQ_60SEC;
+	} else if (val->val1 == 0 && val->val2 == 250000) {
+		mode = CCS811_MODE_IAQ_60SEC;
```
Comment (Mani-Sadhasivam) [inline drivers/sensor/ccs811/ccs811.c]: Should be CCS811_MODE_RAW_DATA?

**Review by Mani-Sadhasivam — CHANGES_REQUESTED** [review]
Looks like you haven't updated the mode in setting!

**Pull request closed without merging**